compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/incremental/HybridNodeStorage.java

```diff
@@ -80,6 +80,9 @@ HybridNode peekSubtree() {
     }
 
     HybridNode consumeToken() {
+        if (currentToken == null) {
+            currentToken = peekToken();
+        }
         HybridNode hybridNode = currentToken;
```

Comment:
Can you please create an issue for this change? I think we need to fix this in other places too. Issue will help us to track it. Thanks.

Reply:
Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/23719